### DISCUSSION TITLE
Extract content of /K

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#1631] [Looking for help]
Can you let me know how the content associated with a particular xref can be extracted?

There is an xref 63 whose object structure is 
![image](https://user-images.githubusercontent.com/36301306/157372446-af3dfc61-232d-4784-bb9e-99edf5eca3e4.png)
Can you let me know what content this xref is referring to?

**Comment by JorjMcKie** [comment]
The object type `StructElem` is explained on page 558 of the PDF [spec](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf). Key `/K` can have several very different meanings according to this.

**Reply by meghanaviyyapu** [reply]
![image](https://user-images.githubusercontent.com/36301306/157408111-c411d349-9e30-4041-8f4f-0e28090447ac.png)
When I went through the content stream of /Pg 3 0 R, I found the above text to which /K is pointing. Can you let me know how this text can be extracted.

In some cases, /K has xrefs like below.
![image](https://user-images.githubusercontent.com/36301306/157408885-de5b36ae-6718-4fe1-a487-3b1c27a5bbde.png)
My objective is to get the content of /StructElem and map it to xref. Can you let me know how this can be achieved?

**Comment by JorjMcKie** [comment]
I have no knowledge / experience of structure elements. But as the description says, `/Pg` is the reference to a PDF page. And it is clear how page text can be extracted.
So the problem is only, how to find the page number belonging to xref 3. As there is no special function for this in PyMuPDF, you must loop through the document's pages and search the page with `page.xref == 3`.

**Reply by meghanaviyyapu** [reply]
Can you let me know if there is a way to extract the text corresponding to one StructElem because there can be multiple StructElems with same page number?

**Reply by JorjMcKie** [reply]
Sorry, I have no idea how to do that.
